### PR TITLE
#6Task. Dzianis Shupenka

### PR DESCRIPTION
**Pull request by dzianisshupenka**
https://travis-ci.org/github/dzianisshupenka/js-assignments/builds/706856901

### CODE REVIEW

File: task/06-conditions-n-loops-tasks.js [hunk 1]
```diff
@@ -46,6 +50,11 @@ function getFizzBuzz(num) {
  *   10 => 3628800
  */
 function getFactorial(n) {
+    let res = 1;
+    while(n){
```
Comment (FedorovD) [inline task/06-conditions-n-loops-tasks.js]: do not use while-loop, using a for-loop is preferred

File: task/06-conditions-n-loops-tasks.js [hunk 2]
```diff
@@ -63,6 +72,11 @@ function getFactorial(n) {
  *   -1,1  =>  0  ( = -1 + 0 + 1 )
  */
 function getSumBetweenNumbers(n1, n2) {
+    let res=0;
+    while(n1<=n2){
```
Comment (FedorovD) [inline task/06-conditions-n-loops-tasks.js]: use for-loop

File: task/06-conditions-n-loops-tasks.js [hunk 3]
```diff
@@ -119,6 +134,13 @@ function isTriangle(a,b,c) {
  *  
  */
 function doRectanglesOverlap(rect1, rect2) {
+    if ((rect1.left + rect1.width) <= rect2.left || (rect1.top + rect1.height) <= rect2.top) { 
+        return false; 
+    }
+    if ((rect2.left + rect2.width ) <= rect1.left || (rect2.top + rect2.height) <= rect1.top) { 
+        return false; 
+    } 
+    return true;
```
Comment (FedorovD) [inline task/06-conditions-n-loops-tasks.js]: there are `else if` and `else`, use them too

File: task/06-conditions-n-loops-tasks.js [hunk 4]
```diff
@@ -209,6 +245,11 @@ function getIntervalString(a, b, isStartIncluded, isEndIncluded) {
  * 'noon' => 'noon'
  */
 function reverseString(str) {
+    let res = '';
+    for (var i = str.length - 1; i >= 0; i--) {
```
Comment (FedorovD) [inline task/06-conditions-n-loops-tasks.js]: `var` is deprecated. It has no sense to use for browsers code even currently.

File: task/06-conditions-n-loops-tasks.js [hunk 5]
```diff
@@ -226,6 +267,12 @@ function reverseString(str) {
  *   34143 => 34143
  */
 function reverseInteger(num) {
+    let str = num.toString();
+    let res = '';
+    for (var i = str.length - 1; i >= 0; i--) {
```
Comment (FedorovD) [inline task/06-conditions-n-loops-tasks.js]: `var` is deprecated. It has no sense to use for browsers code even currently.

**Comment by FedorovD** [comment]
Fix all notes pls and pay attention to js code style
https://learn.javascript.ru/coding-style

I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by dzianisshupenka** [comment]
https://travis-ci.org/github/dzianisshupenka/js-assignments/builds/711680848

**Comment by FedorovD** [comment]
Closing the PR as accepted! Thanks!

**Pull request closed without merging**